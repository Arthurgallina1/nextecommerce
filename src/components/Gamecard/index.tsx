import Image from 'next/image'
import Link from 'next/link'
import Ribbon, { RibbonColors, RibbonSizes } from '../Ribbon'
import CartButton from 'components/CartButton'
import WishlistButton from 'components/WishlistButton'
import { formatPrice } from 'utils/formatters'
import * as S from './styles'

export type GamecardProps = {
  id: string
  slug: string
  title: string
  developer: string
  img: string
  price: number
  promotionalPrice?: number
  favorite?: boolean
  ribbon?: React.ReactNode
  ribbonColor?: RibbonColors
  ribbonSize?: RibbonSizes
  onFav?: () => void
}

const Gamecard = ({
  id,
  slug,
  title,
  developer,
  img,
  price,
  promotionalPrice,
  ribbon,
  ribbonColor = 'primary',
  ribbonSize = 'small'
}: GamecardProps) => (
  <S.Wrapper data-cy="game-card">
    {!!ribbon && (
      <Ribbon color={ribbonColor} size={ribbonSize}>
        {ribbon}
      </Ribbon>
    )}
    <Link href={`/game/${slug}`} passHref>
      <S.ImageBox>
        <Image
          src={'https://via.placeholder.com/150' || img}
          alt={title}
          layout="fill"
        />
      </S.ImageBox>
    </Link>
    <S.Content>
      <Link href={`/game/${slug}`} passHref>
        <S.Info>
          <S.Title>{title}</S.Title>
          <S.Developer>{developer}</S.Developer>
        </S.Info>
      </Link>
      <S.FavButton role="button">
        <WishlistButton id={id} />
      </S.FavButton>
      <S.BuyBox>
        {!!promotionalPrice && (
          <S.Price isPromotional>{formatPrice(price)}</S.Price>
        )}
        <S.Price>
          {price === 0 ? 'FREE' : formatPrice(promotionalPrice || price)}
        </S.Price>
        <CartButton id={id} />
      </S.BuyBox>
    </S.Content>
  </S.Wrapper>
)

export default Gamecard
