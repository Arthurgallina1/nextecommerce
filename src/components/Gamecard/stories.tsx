import { Story, Meta } from '@storybook/react/types-6-0'
import Gamecard, { GamecardProps } from '.'

export default {
  title: 'Gamecard',
  component: Gamecard,
  args: {
    title: 'Population Zero',
    developer: 'Rockstar games',
    img: 'https://source.unsplash.com/user/willianjusten/1042x580',
    price: 'R$ 235,00'
  }
} as Meta

export const Basic: Story<GamecardProps> = (args) => (
  <div style={{ width: '30rem' }}>
    <Gamecard {...args} />
  </div>
)
