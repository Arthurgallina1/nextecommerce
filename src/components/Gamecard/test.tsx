import { render, screen } from '@testing-library/react'

import Gamecard from './index'

describe('Gamecard', () => {
  it('should render the heading', () => {
    const { container } = render(<Gamecard />)
    expect(
      screen.getByRole('heading', { name: /Gamecard/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

})
