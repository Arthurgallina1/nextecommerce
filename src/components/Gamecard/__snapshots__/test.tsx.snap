// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gamecard should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Gamecard
  </h1>
</main>
`;
